### ISSUE TITLE
0.2.1 Error: Cannot read property 'reduce' of undefined

### ISSUE THREAD

**Issue by chrisblossom** [faceyspacey/webpack-flush-chunks#3]
After updating to ``v0.2.1`` the following error occurs. I tested this against the [flush-chunks-boilerplate](https://github.com/faceyspacey/flush-chunks-boilerplate) repo and it is happening there as well.

```js
TypeError: Cannot read property 'reduce' of undefined
    at createFilesByPath (/Users/chris/github/app/build/webpack:/~/webpack-flush-chunks/dist/flushChunks.js:101:1)
    at createFilesByModuleId (/Users/chris/github/app/build/webpack:/~/webpack-flush-chunks/dist/flushChunks.js:116:1)
    at flushWebpack (/Users/chris/github/app/build/webpack:/~/webpack-flush-chunks/dist/flushChunks.js:90:1)
    at flush (/Users/chris/github/app/build/webpack:/~/webpack-flush-chunks/dist/flushChunks.js:70:1)
    at flushChunks (/Users/chris/github/app/build/webpack:/~/webpack-flush-chunks/dist/flushChunks.js:37:1)
    at module.exports.../node_modules/webpack-flush-chunks/dist/flushChunks.js.exports.default (/Users/chris/github/app/build/webpack:/~/webpack-flush-chunks/dist/flushChunks.js:29:1)
    at Html (/Users/chris/github/app/build/webpack:/src/server.js:85:1)
    at /Users/chris/github/app/node_modules/react-dom/lib/ReactCompositeComponent.js:306:16
    at measureLifeCyclePerf (/Users/chris/github/app/node_modules/react-dom/lib/ReactCompositeComponent.js:75:12)
    at ReactCompositeComponentWrapper._constructComponentWithoutOwner (/Users/chris/github/app/node_modules/react-dom/lib/ReactCompositeComponent.js:305:14)
```

I tried to create a test case but was unsuccessful, sorry.

**Comment by faceyspacey** [comment]
fuck, i meant to push last night's commit as a breaking change, so it would upgrade it to 1.0. ...im bout to officially release this today FYI. the API changed. basically its 2 args: `flushChunks(stats, options)`. ...check the source for now. i'll be updating the docs today.

**Comment by chrisblossom** [comment]
Thanks for the quick reply! I figured it was something along those lines. I just wanted to open an issue to make sure you were aware there was something up.

**Comment by chrisblossom** [comment]
Looks like ``moduleIds`` is now an ``option``? At first glance this looks like a mistake (for lack of a better word)?

**Comment by faceyspacey** [comment]
yea, it's the same name for both webpack module ids and babel module paths (which I assume ur using).

**Comment by chrisblossom** [comment]
Sorry I wasn't very clear. ``options`` typically are not part of the input, but rather change the behavior of the calling function. Unless I am mistaken, ``moduleIds`` is part of the input, similar to ``clientStats``.

I could be misunderstanding what ``moduleIds`` are used for.

To me, something more like this would make more sense:
```js
 flushChunks({ moduleIds, clientStats }, {
        before: ['bootstrap'],
        after: ['main'],

        // only needed if serving css rather than an external stylesheet
        outputPath,
    });
```

**Comment by faceyspacey** [comment]
it's 
```js
 flushChunks(clientStats, {
        moduleIds,
        before: ['bootstrap'],
        after: ['main'],
    });
```

you think 3 options makes more sense? 

....u may be right.

i'd just put it back to the old version until a release which i plan to promote later today.

**Comment by faceyspacey** [comment]
my thinking on 2 args was to emphasize the importance of stats, and because it's easier to remember 2 args, with one being options.

**Comment by chrisblossom** [comment]
Just two args still:

This also allows for easier extendibility later on.

```jsx
type ArgsType = {
  moduleIds: Array<string>,
  stats: Object,
  // ...
};

type OptionsType = {
  before?: Array<string>,
  after?: Array<string>,
  outputPath?: string,
  // ...
};

flushChunks(args: ArgsType, options: OptionsType);
```

But the two ``args + options`` before made more sense than having ``moduleIds`` inside ``options`` IMO.

**Comment by faceyspacey** [comment]
...oh, but in the new version it's 2 args, but the first arg isn't an object--it's just the stats. what do u think about that?

...FYI npm is updated with a patch bump back to the old version: https://www.npmjs.com/package/webpack-flush-chunks

**Comment by chrisblossom** [comment]
I was just trying to show you that if you wanted less args, you could make the first arg an object that contains all needed inputs, and the second arg used for options. Make sense?

``stats`` in my example is ``Object`` type because I didn't know exactly what Webpack stats ``type`` is.

**Comment by faceyspacey** [comment]
yea, but if they're named as keys on an object it just might as well be in the options object. i personally dont like 2 options-style arguments. 

i may just put it back to the way it was. that has the benefit of developers not wondering why they are passing babel paths under the name `moduleIds`. thanks for pointing that out.

**Comment by faceyspacey** [comment]
...actually i just realized i cant--i made this decision cuz now u can pass `{ chunkNames }` or `{ moduleIds }` ...i.e. if ur using the new webpack chunknames feature i linked to in our other conversation. ...so `clientStats` is the only remaining stable option. 

im gonna close this. let me know if there's anything else u need.

**Comment by chrisblossom** [comment]
Yeah, that is what I was trying to point out with having the first arg as the ``input args``, and second as ``options``.

**Comment by chrisblossom** [comment]
Another thought is to take [extract-text-webpack-plugin](https://github.com/webpack-contrib/extract-text-webpack-plugin)'s syntax and just accept one arg as an object.

**Comment by faceyspacey** [comment]
here's the article and the release of everything:

https://medium.com/@faceyspacey/code-cracked-for-code-splitting-ssr-in-reactlandia-react-loadable-webpack-flush-chunks-and-1a6b0112a8b8

....our conversation (about Async Reactor and what u said about Next.js getInitialProps) led to some very cool findings in terms of the future.
